test-cargo-miri/subcrate/src/lib.rs

```diff
@@ -1,2 +1,5 @@
+#[cfg(doctest)]
+use num_cpus as _;
```

Comment:
I'm a bit confused by this, I expected to see a doctest instead?
```rust
//! ```rust
//! assert!(true);
//! ```
```

Reply:
I wanted to test that it does not invoke rustdoc at all. The error in #1813 happens when rustdoc is collecting doctests (if I remember correctly), and it fails because it found a dependency was not compiled using the libcore in Miri sysroot (thus I use `use num_cpus` here).

That said, maybe
```rust
#[cfg(doctest)]
compile_error!("rustdoc should not touch me");
```
is clearer.

Reply:
I like the `compile_error!` version.